zcncore/http_consensus_objects.go

```diff
@@ -0,0 +1,91 @@
+package zcncore
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+var ErrNilHttpConsensusObjects = errors.New("nil_httpconsensusmaps")
+
+type HttpConsensusObjects struct {
+	ConsensusThresh int
+	MaxConsensus    int
+
+	WinMap          json.RawMessage
+	WinMapConsensus map[string]int
+
+	WinError string
+	WinInfo  string
+}
+
+func NewHttpConsensusObjects(consensusThresh int) *HttpConsensusObjects {
+	return &HttpConsensusObjects{
+		ConsensusThresh: consensusThresh,
+		WinMapConsensus: make(map[string]int),
+	}
+}
+
+func (c *HttpConsensusObjects) GetValue() (json.RawMessage, bool) {
+	if c == nil {
+		return nil, false
+	}
+	return c.WinMap, true
+}
+
+func (c *HttpConsensusObjects) Add(statusCode int, respBody string) error {
+	if c == nil {
+		return ErrNilHttpConsensusMaps
+	}
+	if statusCode != http.StatusOK {
+		c.WinError = respBody
+		return nil
+	}
+
+	m, hash, err := c.buildObject(respBody)
+	if err != nil {
+		return err
+	}
+
+	c.WinMapConsensus[hash]++
+
+	if c.WinMapConsensus[hash] > c.MaxConsensus {
+		c.MaxConsensus = c.WinMapConsensus[hash]
+		c.WinMap = m
+		c.WinInfo = respBody
+	}
+
+	return nil
+}
+
+func (c *HttpConsensusObjects) buildObject(respBody string) (json.RawMessage, string, error) {
```

Comment:
why can't we resty and already implemented functions for it?

Reply:
We cannot use already existing concensus structure, 'cause it was made for the map-like data. I didn't copy the main logic. Unmarshaling process is completely different

Reply:
https://github.com/0chain/gosdk/blob/e769e25b952657c78986b07943ab9660213b3f8e/zcncore/transaction_query.go#L316 is not capable to be built for mobile devices